lib/std/math/Interval.zig

```diff
@@ -0,0 +1,101 @@
+const std = @import("../std.zig");
+
+/// Create an interval of type T.
+/// See https://en.wikipedia.org/wiki/Interval_(mathematics) for more info.
+pub fn Interval(comptime T: type) type {
+    comptime std.debug.assert(std.meta.trait.isNumber(T)); // Interval must be instantiated with a number type.
+    return struct {
+        /// The left hand side of the Interval.
+        /// n as null here represents infinity.
+        /// open determines if the interval should be open (exclusive) or closed (inclusive). 
+        from: Endpoint,
+        /// The right hand side of the Interval.
+        /// n as null here represent infinity.
+        /// open determines if the interval should be open (exclusive) or closed (inclusive). 
+        to: Endpoint,
+
+        pub const Endpoint = struct {
+            n: ?T,
+            open: bool,
+        };
+
+        pub const Self = @This();
+
+        // TODO: comptime_int and comptime_float support wants self parameter to have the comptime keyword
+        pub fn contains(self: Self, x: T) bool {
+            const min = switch (@typeInfo(T)) {
+                .Int => std.math.minInt(T),
+                // .ComptimeInt => x - 2,
+                .Float => -std.math.inf(T),
+                // .ComptimeFloat => x - 2,
+                else => unreachable,
+            };
+            const max = switch (@typeInfo(T)) {
+                .Int => std.math.maxInt(T),
+                // .ComptimeInt => x + 2,
+                .Float => std.math.inf(T),
+                // .ComptimeFloat => x + 2,
+                else => unreachable,
+            };
+            if (self.from.n) |_| {} else {
+                // handle (-inf,
+                const to = self.to.n orelse max;
+                const right = if (self.to.open) x < to else x <= to;
+                return right;
+            }
+            if (self.to.n) |_| {} else {
+                // handle +inf)
+                const from = self.from.n orelse min;
+                const left = if (self.from.open) x > from else x >= from;
+                return left;
+            }
+            const from = self.from.n orelse min;
+            const to = self.to.n orelse max;
+            const left = if (self.from.open) x > from else x >= from;
+            const right = if (self.to.open) x < to else x <= to;
+            return left and right;
+        }
+    };
+}
+
+//
```

Comment:
Not sure why this comment is hanging out here?

Reply:
I added this when I was working on it to separate the public API/implementation from the tests. If this is against style, I can remove it.